pkg/kinflate/app/application.go

```diff
@@ -17,18 +17,26 @@ limitations under the License.
 package app
 
 import (
-	"k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
-	"k8s.io/kubectl/pkg/kinflate/resource"
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	interror "k8s.io/kubectl/pkg/kinflate/internal/error"
+	"k8s.io/kubectl/pkg/kinflate/types"
 	"k8s.io/kubectl/pkg/loader"
 )
 
 type Application interface {
-	Resources() []resource.Resource
+	// Resources returnsn the transformed resources.
+	Resources() (types.ResourceCollection, error)
+	// RawResources computes and returns the raw resources from the manifest.
+	// It contains resources from 1) untransformed resources from current manifest 2) transformed resources from sub packages
+	RawResources() (types.ResourceCollection, error)
 }
 
+var _ Application = &applicationImpl{}
+
 // Private implementation of the Application interface
 type applicationImpl struct {
-	manifest v1alpha1.Manifest
+	// manifest is used for getting transformer.
```

Comment:
I don't think this comment is useful. may be omit it completely. manifest is everything for an app  :)